toplevel/coqcargs.ml

```diff
@@ -62,26 +62,29 @@ let error_missing_arg s =
   prerr_endline "See -help for the syntax of supported options";
   exit 1
 
-let check_compilation_output_name_consistency args =
-  match args.compilation_output_name, args.compile_list with
-  | Some _, _::_::_ ->
-    prerr_endline ("Error: option -o is not valid when more than one");
-    prerr_endline ("file have to be compiled")
-  | _ -> ()
+let arg_error msg = CErrors.user_err msg
 
 let is_dash_argument s = String.length s > 0 && s.[0] = '-'
 
 let add_compile ?echo copts s =
-  if is_dash_argument s then (prerr_endline ("Unknown option " ^ s); exit 1);
-  (* make the file name explicit; needed not to break up Coq loadpath stuff. *)
-  let echo = Option.default copts.echo echo in
-  let s =
-    let open Filename in
-    if is_implicit s
-    then concat current_dir_name s
-    else s
-  in
-  { copts with compile_list = (s,echo) :: copts.compile_list }
+  match s, copts.compile_file with
+  | extra :: _ :: _, _
```

Comment:
This is some really perplexing code. I think that, for the sanity of all of us, the function should be changed so that it takes only one file rather than a list. Then the code at line 235 would be using `List.fold_left` to call the function.

Reply:
Yes, I was lazy sorry, will improve.